src/configuration/apiConfigurationParser.ts

```diff
@@ -10,8 +11,18 @@ export class DefaultApiConfigurationParser implements ApiConfigurationParser {
     parse(xmlConfig: string): Promise<ApiConfiguration> {
         return new Promise((resolve, reject) => {
             parseString(xmlConfig, function (err, result) {
-                if (err) reject(err);
-                else resolve(new DefaultApiConfiguration(result));
+                if (err) {
+                    reject(err);
+                }
+                else {
+                    const rawConfig = result as RawApiConfiguration;
+                    if (rawConfig) {
+                        resolve(new DefaultApiConfiguration(result));
+                    }
+                    else {
+                        reject(new Error(`invalid configuration: ${xmlConfig}`));
```

Comment:
you probably need to catch the exception raised by DefaultApiConfig in case the config is not valid.